external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/CachedKafkaConsumer.scala

```diff
@@ -83,6 +86,139 @@ private[kafka010] case class CachedKafkaConsumer private(
     record
   }
 
+  /**
+   * Get the record at the `offset`. If it doesn't exist, try to get the earliest record in
+   * `[offset, untilOffset)`.
+   */
+  def getAndIgnoreLostData(
+      offset: Long,
+      untilOffset: Long,
+      pollTimeoutMs: Long): ConsumerRecord[Array[Byte], Array[Byte]] = {
+    // scalastyle:off
+    // When `failOnDataLoss` is `false`, we need to handle the following cases (note: untilOffset and latestOffset are exclusive):
```

Comment:
Also is beginningOffset the earlier offset available in Kafka? Then lets rename this as earliestOffset, and "earliest" and "latest" and more popular terms in the context of Kafka